gluten-core/src/main/scala/org/apache/gluten/extension/columnar/OffloadSingleNode.scala

```diff
@@ -181,7 +159,24 @@ case class OffloadJoin() extends OffloadSingleNode with LogLevelUtil {
       case other => other
     }
   }
+}
 
+object OffloadJoin {
+  def transformSortMergeJoinExec(smj: SortMergeJoinExec): TransformSupport = {
+    if (GlutenConfig.getConf.forceShuffledHashJoin) {
```

Comment:
Can we add a new rule to convert smj to shj first ? There are many code place check `forceShuffledHashJoin`. I'm afraid there may be some leaked and other developers also may miss it in future.